lib/cpp/prox/prox_separable.cpp

```diff
@@ -74,61 +77,70 @@ double ProxSeparable::call_single(double x,
 }
 
 // Compute the prox on the i-th coordinate only
-void ProxSeparable::call_single(ulong i,
-                                const ArrayDouble &coeffs,
-                                double step,
-                                ArrayDouble &out) const {
+template <class T>
+void TProxSeparable<T>::call_single(ulong i, const Array<T> &coeffs, T step,
+                                    Array<T> &out) const {
   if (i >= coeffs.size()) {
-    TICK_ERROR(get_class_name() << "::call_single " << "i= " << i << " while coeffs.size()=" << coeffs.size());
+    TICK_ERROR(get_class_name()
+               << "::call_single "
+               << "i= " << i << " while coeffs.size()=" << coeffs.size());
   } else {
     if (has_range) {
       if ((i >= start) && (i < end)) {
-        out[i] = call_single(coeffs[i], step);
+        set_out_i(out, i, call_single(coeffs[i], step));
```

Comment:
why do we introduce and use a set_ou_i ? And why don't we use it before ? (such as in void TProxSeparable<T>::call(const Array<T> &coeffs, const Array<T> &step, Array<T> &out, ulong start, ulong end))

Reply:
yes good catch! it's not needed yet